[dynamo][nn_module] Replace UserFunctionVariable with VariableTracker build

anijain2305 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #165896
* #165707
* __->__ #165708

Audit: To prevent future issues with functools.partial or callable objects.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @Lucaskabela

anijain2305 committed:
[dynamo][nn_module] Replace UserFunctionVariable with VariableTracker build

Audit: To prevent future issues with functools.partial or callable objects.

[ghstack-poisoned]

Lucaskabela reviewed (approved):
Not sure what the missing regression test signal is from, but I think it is fine to ignore (although if you are familiar with it maybe add)

anijain2305 committed:
Update on "[dynamo][nn_module] Replace UserFunctionVariable with VariableTracker build"


Audit: To prevent future issues with functools.partial or callable objects.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo][nn_module] Replace UserFunctionVariable with VariableTracker build"


Audit: To prevent future issues with functools.partial or callable objects.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo][nn_module] Replace UserFunctionVariable with VariableTracker build"


Audit: To prevent future issues with functools.partial or callable objects.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo][nn_module] Replace UserFunctionVariable with VariableTracker build"


Audit: To prevent future issues with functools.partial or callable objects.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]

anijain2305 commented:
@pytorchbot merge
